src/drivers/nuopc/shr/seq_flds_mod.F90

```diff
@@ -541,1440 +541,1056 @@ subroutine seq_flds_set(nmlfile, ID, cime_model)
      ! domain coordinates
      !----------------------------------------------------------
 
-     call seq_flds_add(dom_coord,'lat')
      longname = 'latitude'
      stdname  = 'latitude'
      units    = 'degrees north'
-     attname  = 'lat'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(dom_coord,'lat', longname, stdname, units)
 
-     call seq_flds_add(dom_coord,'lon')
      longname = 'longitude'
      stdname  = 'longitude'
      units    = 'degrees east'
-     attname  = 'lon'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(dom_coord,'lon', longname, stdname, units)
 
-     call seq_flds_add(dom_coord,'hgt')
      longname = 'height'
      stdname  = 'height, depth, or levels'
      units    = 'unitless'
-     attname  = 'hgt'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(dom_coord,'hgt', longname, stdname, units)
 
-     call seq_flds_add(dom_other,'area')
      longname = 'cell_area_model'
      stdname  = 'cell area from model'
      units    = 'radian^2'
-     attname  = 'area'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(dom_other,'area', longname, stdname, units)
 
-     call seq_flds_add(dom_other,'aream')
      longname = 'cell_area_mapping'
      stdname  = 'cell area from mapping file'
      units    = 'radian^2'
-     attname  = 'aream'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(dom_other,'aream', longname, stdname, units)
 
-     call seq_flds_add(dom_other,'mask')
      longname = 'mask'
      stdname  = 'mask'
      units    = '1'
-     attname  = 'mask'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(dom_other,'mask', longname, stdname, units)
 
-     call seq_flds_add(dom_other,'frac')
      longname = 'area_fraction'
      stdname  = 'area fraction'
      units    = '1'
-     attname  = 'frac'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(dom_other,'frac', longname, stdname, units)
 
      !----------------------------------------------------------
      ! states/fluxes from atm
      !----------------------------------------------------------
 
      ! height at the lowest model level (m)
-     call seq_flds_add(a2x_states,"Sa_z")
-     call seq_flds_add(x2l_states,"Sa_z")
-     call seq_flds_add(x2i_states,"Sa_z")
      longname = 'Height at the lowest model level'
      stdname  = 'height'
      units    = 'm'
-     attname  = 'Sa_z'
-     call metadata_set(attname, longname, stdname, units)
+     call seq_flds_add(a2x_states,"Sa_z", longname, stdname, units)
```

Comment:
Optional arguments should always appear as keyword/value pairs. This is not consistent in the code and should be fixed.